packages/common/e2e/dynamic-value-resolver.e2e.ts

```diff
@@ -0,0 +1,226 @@
+    factory = new Artifactory(config);
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 200000;
+  });
+  it("create and destroy harness", async () => {
+    const ctxMgr = await factory.getContextManager(orgName, "admin");
+    const items = await createItems(5, false, ctxMgr.context);
+
+    try {
+      // given the initiative, resolve it's dynamic values
+      const start = Date.now();
+      const resolved = await resolveEntityDynamicValues(
+        items.initiative.toJson(),
+        ctxMgr.context
+      );
+      const end = Date.now();
+      const duration = end - start;
+
+      // expectations
+      const expected = {
+        treesPlanted: 60,
+        statusCounts: {
+          started: 2,
+          planned: 2,
+        },
+        surveyCounts: 204,
+        orgShort: "qa-pre-hub",
+        orgName: "QA Premium Hub",
+        extent: {
+          xmin: -14999999.999999745,
+          ymin: 2699999.999999951,
+          xmax: -6199999.999999896,
+          ymax: 6499999.99999985,
+          spatialReference: {
+            wkid: 102100,
+          },
+        },
+      };
+
+      expect(resolved.orgShort).toEqual(expected.orgShort, "orgShort mismatch");
+      expect(resolved.orgName).toEqual(expected.orgName, "orgName mismatch");
+      expect(resolved.extent).toEqual(expected.extent, "extent mismatch");
+
+      // console.info(
+      //   `Resolved for ${items.projects.length} children in ${duration}ms`
+      // );
+      // console.info(`Static Values took ${timings.staticValues}ms`);
+      // console.info(`Query Values took ${timings.queryValues}ms`);
+    } catch (ex) {
+      expect(ex).toBeNull('Exception thrown: "' + ex + '"');
+    } finally {
+      await cleanupItems(items, ctxMgr.context);
+    }
+  });
+});
+
+const timings: Record<string, number> = {
+  total: 0,
+  staticValues: 0,
+  queryValues: 0,
+};
+
+async function resolveEntityDynamicValues(
+  initiative: IHubInitiative,
+  context: IArcGISContext
+): Promise<Record<string, any>> {
+  const dv = (initiative.dynamicValues ||
+    []) as unknown as DynamicValueDefinition[];
+  return resolveDynamicValues(dv, context);
+}
+
+interface ICreateOutput {
+  initiative: HubInitiative;
+  projects: HubProject[];
+}
+
+async function createItems(
+  childCount: number,
+  includeFSQuery: boolean,
+  context: IArcGISContext
+): Promise<ICreateOutput> {
+  const result: ICreateOutput = {
+    // THIS IS TERRIBLE! DO NOT DO THIS IN PRODUCTION CODE!
+    initiative: null as unknown as HubInitiative,
+    projects: [],
+  };
+  // Create an initiative
+  const init: Partial<IHubInitiative> = {
+    name: "Test Initiative",
+    description: "This is a test initiative",
+    orgUrlKey: context.portal.urlKey,
+  };
+  const instance = await HubInitiative.create(init, context, true);
+  // Construct typed dynamic value definitions
+  const treesDynamicValue: IDynamicItemQueryDefinition = {
+    source: "item-query",
+    sourcePath: "properties.values.treesPlanted",
```

Comment:
similar to my comment on functions for search options - can we encapsulate the `sourcePath` with a function like `getDynamicQuerySourcePath( 'treesPlanted' )` which removes developer needing to know if we use `properties.value` or if we changed later to storing another way